internal/services/compute/image_resource.go

```diff
@@ -143,6 +143,14 @@ func resourceImage() *pluginsdk.Resource {
 							ForceNew:     true,
 							ValidateFunc: validate.DiskEncryptionSetID,
 						},
+
+						"storage_type": {
```

Comment:
Unfortunately this property has a default value (`Standard_LRS`), so users will get a diff if their image does not use it, and will result in a destroy/recreate. (e.g. if the user has imported the resource, or does not add the default value to their config.)
Since we have a major version release in the near future, you can put this schema change and related code behind the 4.0 beta flag if you add the `Default`, then this property can be available in the beta opt-int, and will ship fully in 4.0.

Since you appear to have updated all the test configs, this would be hidden in testing, if you remove setting it from one of the tests, you should see what I mean.